### PR TITLE
feat(headless-styles): convert toast to scss

### PR DESCRIPTION
**Pull request by Rykus0**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

contributes to #1242 

## What is the new behavior?
Converts Toast to SCSS

## Other information
[Sandbox](https://codesandbox.io/s/ps-react-typescript-forked-tbk3ez)

### CODE REVIEW

File: packages/headless-styles/src/components/Toast/toastCSS.scss [hunk 1]
```diff
@@ -73,26 +74,30 @@
 
 /* sentiment */
 
-.successToast {
-  composes: toastBase;
+.pando_successToast {
```
Comment (caseybaggz) [inline packages/headless-styles/src/components/Toast/toastCSS.scss:77]: We should be able to use an iterable mixin here too and just use a sentiment map. Probably even want to make that in shared so we can always iterate over sentiments when needed? 🤔

Comment (caseybaggz) [reply packages/headless-styles/src/components/Toast/toastCSS.scss:77]: Something like `--ps-${sentiment}-surface` if it's possible in sass?

Comment (Rykus0) [reply packages/headless-styles/src/components/Toast/toastCSS.scss:77]: Definitely possible, and I think something we could get some good usage out of.

File: packages/headless-styles/src/components/shared/_tokens.scss [hunk 2]
```diff
@@ -0,0 +1,15 @@
+@use 'sass:list';
+
+$sentiments: 'default', 'action', 'info', 'success', 'warning', 'danger';
+
+@function sentiments-except($excludedSentiments...) {
```
Comment (caseybaggz) [inline packages/headless-styles/src/components/shared/_tokens.scss:5]: Reading this is pretty hard. I actually think the TS methods are pretty nice and think we should use that same design. So, I would do a couple of things here.

1. Rename this file to be `_vars` and keep the sentiment variable in here
2. Create a new file called `_utils` and move the function in there and make 2 different functions called `pick` and `exclude` so we have the choice for whichever is easier. Params should match the TS design of the `list` to choose from and the vars you want to pick/exclude

**Review by caseybaggz — APPROVED** [review]

**Pull request merged**